pennylane/qnode.py

```diff
@@ -670,15 +670,21 @@ def __call__(self, *args, **kwargs):
             # construct the tape
             self.construct(args, kwargs)
 
-        # If the observable contains a Hamiltonian and the device does not
-        # support Hamiltonians, or if the simulation uses finite shots,
-        # split tape into multiple tapes of diagonalizable known observables.
-        # In future, this logic should be moved to the device
-        # to allow for more efficient batch execution.
+        # Under certain conditions, split tape into multiple tapes and recombine them.
+        # Else just execute the tape, and let the device take care of things.
+        # TODO (future squad): This logic should be moved away from the qnode, preferably to the device.

```

Comment:
```suggestion

```
Might remove this, since it's been done (all that's left to do is 'delete' this QNode).